### ISSUE TITLE
electron11 build JSON error

### ISSUE THREAD

**Issue by ChangJi** [electron/electron#25438]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
11.0.0-beta.1
* **Operating System:**
 macOS 10.15.6
* **Last Known Working Electron version:**
11.0.0-beta.1

### Expected Behavior
No Error

### Actual Behavior

Uncaught Exception:
SyntaxError: Error parsing /Applications/electron11.app/Contents/Resources/app.asar/node_modules/ffmpeg-static/package.json: Unexpected end of JSON input

### To Reproduce
packages:
 "electron-builder": "^22.8.0",
 "ffmpeg-static": "^4.2.0",

in main.js ：
var pathToFfmpeg = require('ffmpeg-static');
console.log('pathToFfmpeg ',pathToFfmpeg);  

npm run build 

install dmg  and open the app

### Screenshots

![image](https://user-images.githubusercontent.com/7902399/92990924-a4200600-f512-11ea-8329-d126c5a31fb4.png)

**Comment by vladimiry** [comment]
I'm facing the same issue on Linux starting from electron v11/beta. The difference is that in my case the native module is `sodium-native`. The error only occurs when the app is packaged and then installed as usually (no issue if I run the app in dev mode). So I guess the issue might be related to asar archiving.
```
A JavaScript error occurred in the main process
Uncaught Exception:
SyntaxError: Error parsing /opt/ElectronMail/resources/app.asar/node_modules/sodium-native/package.json: Unexpected end of JSON input
    at parse (<anonymous>)
    at readPackage (internal/modules/cjs/loader.js:269:20)
    at readPackageExports (internal/modules/cjs/loader.js:309:15)
    at applyExports (internal/modules/cjs/loader.js:450:20)
    at resolveExports (internal/modules/cjs/loader.js:513:23)
    at Function.Module._findPath (internal/modules/cjs/loader.js:641:31)
    at Module._resolveFilename (internal/modules/cjs/loader.js:959:27)
    at Function.n._resolveFilename (electron/js2c/browser_init.js:249:919)
    at Module._load (internal/modules/cjs/loader.js:848:27)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12694)
```

So the issue occurs on at least macOs and Linux systems.

**Comment by ChangJi** [comment]
may be someting wrong in  app.asar.unpacked, the native module url should direct app.asar.unpacked ...?

**Comment by codebytere** [comment]
This looks potentially to be an issue with `electron-builder` and/or https://github.com/electron/electron/issues/18397, so i'm going to close this out in favor of comments in either of those two places, please.

**Comment by vladimiry** [comment]
Referencing the related electron-builder issue https://github.com/electron-userland/electron-builder/issues/5360.

**Comment by mn4367** [comment]
This is not a problem with electron-builder, it also happens with regular non-native modules like fs-extra and using electron-packager alone. The problem occurs here just by switching from electron 10.1.6 to Electron 11.0.3 without changing anything else.

![app asar unpacked](https://user-images.githubusercontent.com/6379696/101025027-40d3cc00-3575-11eb-92ac-8155587fef29.png)

**Comment by vladimiry** [comment]
> This is not a problem with electron-builder

To support this statement I think you might want to pack your app's resources using https://www.npmjs.com/package/asar and see if it works since electron-builder using custom [asar packing implementation](https://github.com/electron-userland/electron-builder/tree/master/packages/app-builder-lib/src/asar) which might potentially be causing the issue.

**Comment by mn4367** [comment]
@vladimiry I'm not sure if I got this right. I'm not using electron-builder, only electron-packager. But just for the record I packed my resources manually with [asar](https://www.npmjs.com/package/asar) from npmjs and compared the output to the asar file which electron-packager created. They are _identical_. I did this with both 10.1.6 and 11.0.3.

Just for completeness, the only difference between both manually packed asar files is the electron version string which is located in a package-lock.json file which was also picked up by asar. Comparing the asar files created by electron-packager leads to the same result, they only differ in the mentioned version string (naturally).

In my case I'm excluding `node_modules` from the asar package. This seems to work, with both 10.1.6 and 11.0.3 electron-packager puts the modules in `app.asar.unpacked` as expected. But only Electron 11.0.3 no longer seems to be able to require a module from there.

I think something has changed in 11.0.3 with regard to accessing files from `app.asar.unpacked` although I didn't find anything related in the release notes. If I don't exclude node_modules from asar Electron 11.0.3 has no problem loading them, but that's not what I want and I suspect others also need to put portions of their resources outside the asar file.

**Comment by vladimiry** [comment]
> I'm not using electron-builder, only electron-packager

I thought you were using electron-builder since you referenced it so I suggested you to try asar npm package. The electron-packager bythe way packs things using asar npm package but electron-builder uses custom implementation.

**Comment by vladimiry** [comment]
@mn4367 I can confirm that it's not only anbout native modules but about referencing anything from the `app.asar.unpacked` dir. So new issues needs to be open since in my experience the closed issues never gets reopened here regardless the reason it was closed.

**Comment by mn4367** [comment]
> PS going to place a new issue soon.

Thanks!

**Comment by vladimiry** [comment]
@mn4367 see https://github.com/electron/electron/issues/26819

**Comment by mn4367** [comment]
@vladimiry Thank you very much!

**Comment by dac09** [comment]
I've verified that this has been fixed in the latest version `11.1.0`. We can enable asar again 👍
